sklearn/utils/testing.py

```diff
@@ -339,13 +339,13 @@ def __repr__(self):
 
     def __enter__(self):
         clean_warning_registry()  # be safe and not propagate state + chaos
-        warnings.simplefilter("ignore", self.category)
         if self._entered:
             raise RuntimeError("Cannot enter %r twice" % self)
         self._entered = True
         self._filters = self._module.filters
         self._module.filters = self._filters[:]
         self._showwarning = self._module.showwarning
+        warnings.simplefilter("ignore", self.category)
```

Comment:
Can you explain why you moved it here?

Should there not be an equivalent `warnings.filters.pop()` in `__exit__`?

Reply:
Suppose there are some external warning filters and local warning filters, and this change means that in the local domain, both the external warning filters and local warning filters are working, and upon exiting this local domain, the external warning filters are restored to be the way they were (before entering the local domain). This is intended to match the behavior of context manager as decorator, since the decorator use `with warnings.catch_warnings():` to do so.
I have proposed to use this in the previous posts several times, I think this makes sense but no one confirms this, and then I found numpy's [suppress_warnings](https://github.com/numpy/numpy/blob/00b7d4ab73f8f394b771de2bbab8a3a4acdbb635/numpy/testing/_private/utils.py#L1946-L1948) has the same behavior.